### DISCUSSION TITLE
Failed to run: ErrNoMore on initial inbox snapshot

### DISCUSSION THREAD

**Discussion by alexklklkl** [Davincible/goinsta#4] [General]
Hello, David, thank you for your work for **goinsta** project! 
Could you help with insta.Export - goinsta.Import? Here is code:

![Снимок экрана 2021-08-22 в 8 29 08](https://user-images.githubusercontent.com/85739808/130344899-5cad8187-df68-41ee-8ab8-ad376004e3df.png)

First launch _(file sessionFileName does not exists)_ all work as expected (but there are error _"Failed to fetch initial messages inbox snapshot: No more posts availible, page end has been reached"_ on **err = insta.Login()**)

Second launch _(file sessionFileName does exists)_ i have error 403 "You've Been Logged Out" in **func (account *Account) Sync()**.

If i use boolean parameter in **goinsta.Import(sessionFileName, true)**, error 403 occurs later. 

File sessionFileName look like correct, only fields "token", "header_options", "cookies" are empty.

What do you think about this error?

**Comment by Davincible** [comment]
Hey! Sorry for the late reply, I didn't get any notification. 

Are you still experiencing this error? I fixed some inbox errors previously. 

If you do I'll have a deeper look at it.

As for the options, the token and cookie field don't matter for this version, only the header_options should contain a bunch of values

**Reply by alexklklkl** [reply]
Yes, it is stable error :)

This fragment:
```
insta = goinsta.New(config.Username, config.Password)
err = insta.Login()
if err != nil {
    panic(err)
}
insta.Export(sessionFileName)
// Next work ok, login is correct
profile, err := insta.VisitProfile(account)
...
```
saves file with empty map for **header_options**:
```
{ ... ,"xmid_expiry":1630589689,"header_options":{},"cookies":null, ... }
```
May be error is in **insta.Export** function.

**Reply by Davincible** [reply]
Fixed now! Thanks for reporting

**Comment by Davincible** [comment]
@alexklklkl double checked and it should be working fine now. On a side note, the incomplete values still got exported because you only printed out the error, while its better to pass high-level errors such as in a main file to .e.g `panic(err)` or `log.Fatal(err)`.

**Comment by alexklklkl** [comment]
Once more interesting case:

![Снимок экрана 2021-09-01 в 20 47 42](https://user-images.githubusercontent.com/85739808/131905600-34a18872-819d-4cfd-961a-0501cf5ef1c9.png)

We'd like to get insta.Timeline.err.Error(), but insta.Timeline.err is nil.

**Reply by Davincible** [reply]
Made some changes for this too, if you still encounter any issue like this please let me know. Thanks for the clear error screenshot! Helps a lot

**Comment by Davincible** [comment]
@alexklklkl Alright, looking at both now
Edit: Fixed

**Reply by alexklklkl** [reply]
Ou, super, header_options filled (use **config** after **set headers**?). Thank you!
```
{..."header_options":{"Authorization":"Bearer IGT:2:ey....}
```
And export needs to be defered... Good idea.

**Reply by Davincible** [reply]
Yess. Thanks for raising the issues. If you have any more issues or ideas about the project please let me know
